File: frontend-mealsonwheels/src/types/user/RegisterDTO.ts
// src/types/user/RegisterDTO.ts

export interface RegisterDTO {
  username: string;
  phoneNumber: string;
  email: string;
  password: string;
  roleName: string;

  // Member-specific
  address?: string;
  dietaryRestriction?: string;

  // Volunteer-specific
  availability?: string;
  services?: string;

  // Rider-specific
  driverLicense?: string;

  // Caregiver-specific
  qualificationAndSkills?: string;

  // Partner-specific
  companyName?: string;
  companyDescription?: string;
  companyAddress?: string;

  // Supporter-specific
  supportType?: string;
  supdescription?: string;

  // Donor-specific
  donorType?: string;
  donationAmount?: number;
}

export type RoleName =
  | 'ROLE_ADMIN'
  | 'ROLE_MEMBER'
  | 'ROLE_VOLUNTEER'
  | 'ROLE_CAREGIVER'
  | 'ROLE_RIDER'
  | 'ROLE_PARTNER'
  | 'ROLE_SUPPORTER'
  | 'ROLE_DONOR';

File: backend-mealsonwheels/src/main/java/com/merrymeal/mealsonwheels/dto copy done/RegisterDTO.java
package com.merrymeal.mealsonwheels_backend.dto;

import lombok.*;
import java.math.BigDecimal;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class RegisterDTO {
    private String username;
    private String phoneNumber;
    private String email;
    private String password;
    private String roleName;

    // Member-specific
    private String address;
    private String dietaryRestriction;

    // Volunteer-specific
    private String availability;
    private String services;

    // Rider-specific
    private String driverLicense;

    // Caregiver-specific
    private String qualificationAndSkills;

    // Partner-specific
    private String companyName;
    private String companyDescription;
    private String companyAddress;

    // Supporter-specific
    private String supportType;
    private String supdescription;

    // Donor-specific
    private String donorType;
    private BigDecimal donationAmount;
}
